test/python/circuit/library/test_grover_operator.py

```diff
@@ -125,9 +144,10 @@ def test_custom_zero_reflection(self):
             expected.h(0)  # state_in is H
             expected.compose(zero_reflection, inplace=True)
             expected.h(0)
-            self.assertEqual(expected, grover_op.decompose())
+            self.assertEqual(expected, grover_op if use_function else grover_op.decompose())
```

Comment:
This tests makes me think that we should mention in the reno and docstring that the output of the new constructor is a series of gates instead of a "black box". This is actually the main user-facing difference I see (besides the mcx detail)